execute/table.go

```diff
@@ -36,43 +36,79 @@ func GroupKeyForRowOn(i int, cr flux.ColReader, on map[string]bool) flux.GroupKe
 	return NewGroupKey(cols, vs)
 }
 
-// OneTimeTable is a Table that permits reading data only once.
-// Specifically the ValueIterator may only be consumed once from any of the columns.
-type OneTimeTable interface {
-	flux.Table
-	onetime()
-}
-
-// CacheOneTimeTable returns a table that can be read multiple times.
-// If the table is not a OneTimeTable it is returned directly.
-// Otherwise its contents are read into a new table.
-func CacheOneTimeTable(t flux.Table, a *memory.Allocator) (flux.Table, error) {
-	_, ok := t.(OneTimeTable)
-	if !ok {
-		return t, nil
-	}
-	return CopyTable(t, a)
+type tableBuffer struct {
+	key     flux.GroupKey
+	colMeta []flux.ColMeta
+	i       int
+	buffers []flux.ColReader
```

Comment:
I get what's happening, and why this works but I'm also concerned.  The purpose (though rarely realized) for the `flux.ColReader`  is to chunk data that is too large to fit into memory.  

Given that most of our streams end up holding all of the data in memory anyway, this is perhaps not a big issue.  But you may want to add a TODO about how this tableBuffer may not be sufficient on massive data sets (if that's even a design goal still, idk).  

Another suggestion would be to make it an `memoryTableBuffer` to make it clearer that the data will be buffered wholly in memory.

Reply:
I agree with this, but I figure that this will be a general consequence of certain operations. I think we included the allocator to catch when the user tries to allocate too much memory and we're going to have to be prepared for the times when a user asks us to buffer a large dataset in memory. I'll add a comment describing this though.